### PR TITLE
get_test_data auto-updates

### PR DESCRIPTION
**Pull request by rheacangeo**
Currently running 'make get_test_data' only checks whether you have data, not whether it is up to date with the FORTRAN_SERIALIZED_DATA_VERSION set in the Makefile. This does not work for those who maintain their git repos in place, as when the version updates upon a git pull, the data will not update. . 
This adds a simple check that md5sums of the local test data matches that in the cloud. If not, the local data is removed and the data version specified in the Makefile is downloaded.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -175,7 +175,8 @@ push_python_regressions:
 	gsutil -m cp -r $(TEST_DATA_HOST)/python_regressions $(REGRESSION_DATA_STORAGE_BUCKET)/$(FORTRAN_SERIALIZED_DATA_VERSION)/$(EXPERIMENT)/python_regressions
 
 get_test_data:
-	if [ ! -f "$(TEST_DATA_HOST)/input.nml" ]; then \
+	if [ ! -f "$(TEST_DATA_HOST)/input.nml" ] || [ "$$(gsutil cat $(REGRESSION_DATA_STORAGE_BUCKET)/$(FORTRAN_SERIALIZED_DATA_VERSION)/${EXPERIMENT}/md5sums.txt)" != "$$(cat $(TEST_DATA_HOST)/md5sums.txt)"  ]; then \
```
Comment (jdahm) [inline Makefile:178]: is it possible to break this onto shorter lines and use backslashes?

Comment (rheacangeo) [reply Makefile:178]: not sure -- Makefile syntax is uber picky, but it can probably be broken into multiple lines

**Comment by ofuhrer** [comment]
Would a rsync instead of a copy work better?

**Comment by rheacangeo** [comment]
Rsync would work better if the data was not tarred, but I don't think it necessarily helps with the small number of files. I could be wrong. rsync has been a little finnicky for me.

**Comment by jdahm** [comment]
rsync only copies the diffs, which is nice. Either way...

**Comment by rheacangeo** [comment]
yep that's true, though I'd rather get this in then go through another cycle of testing if that ok with you?

**Comment by rheacangeo** [comment]
wait we do already use rsync. the cp you are seeing is for the python regressions, unrelated to the this PR, and not in use right now. In that case the amount of data was so tiny it didn't really matter whether cp or rsync were used. If we revive that approach, we should probably just store the regressions in github like the fortran repo does.

File: Makefile [hunk 2]
```diff
@@ -171,13 +173,15 @@ run_tests_parallel:
...
+	gsutil -m cp -r $(TEST_DATA_HOST)/python_regressions $(DATA_BUCKET)python_regressions
 
 get_test_data:
-	if [ ! -f "$(TEST_DATA_HOST)/input.nml" ]; then \
+	if [ ! -f "$(TEST_DATA_HOST)/input.nml" ] || \
+	[ "$$(gsutil cat $(DATA_BUCKET)md5sums.txt)" != "$$(cat $(TEST_DATA_HOST)/md5sums.txt)"  ]; then \
+	rm -rf $(TEST_DATA_HOST) && \
 	$(MAKE) sync_test_data && \
 	$(MAKE) unpack_test_data ;\
 	fi
```
Comment (jdahm) [inline Makefile:187]: Would be cool to indent blocks like this for readability next time we touch the Makefile.

**Review by jdahm — APPROVED** [review]
Looks good. This is going to save us a lot of hassle!

**Pull request merged**